### PR TITLE
Fixes for swig-4.1.0 macro definition correction

### PR DESCRIPTION
**Pull request by wsfulton**
For swig-4.1.0 change:
```
    #2193 -DFOO on the SWIG command line now sets FOO to 1 for
    consistency with C/C++ compiler preprocessors.  Previously
    SWIG set FOO to an empty value.
```
See https://github.com/swig/swig/issues/2193

This patch is backwards compatible with older versions of SWIG.

### CODE REVIEW

**Comment by wsfulton** [comment]
BTW, I can't see the point of these macros as I can't see them being used anywhere in the repository. Perhaps they should be removed altogether?

**Comment by fbergmann** [comment]
@wsfulton  these defines are necessary, when compiling against dependency libraries using C++ namespaces. So they cant be removed. I wonder whether the define should be set to =1 instead.

File: copasi/bindings/perl/CMakeLists.txt [hunk 1]
```diff
@@ -72,13 +72,13 @@ ADD_CUSTOM_COMMAND(
...
-          -DLIBSBML_CPP_NAMESPACE_BEGIN
-          -DLIBSBML_CPP_NAMESPACE_END
-          -DLIBSEDML_CPP_NAMESPACE_USE
-          -DLIBSEDML_CPP_NAMESPACE_BEGIN
-          -DLIBSEDML_CPP_NAMESPACE_END
-          -DLIBSEDML_CPP_NAMESPACE_QUALIFIER
+          -DLIBSBML_CPP_NAMESPACE_USE=
+          -DLIBSBML_CPP_NAMESPACE_BEGIN=
+          -DLIBSBML_CPP_NAMESPACE_END=
+          -DLIBSEDML_CPP_NAMESPACE_USE==
```
Comment (fbergmann) [inline copasi/bindings/perl/CMakeLists.txt:78]: typo

**Comment by fbergmann** [comment]
i'll test in the morning against one of the namespaced libraries and let you know how things go

**Pull request merged**